app/questionnaire/placeholder_parser.py

```diff
@@ -113,6 +101,16 @@ def _parse_transforms(self, transform_list: Sequence[Mapping]):
 
         return transformed_value
 
+    def _resolve_value_source_list(self, value_source_list: list[Dict]) -> list[str]:
+        answers: list[str] = []
+        for value_source in value_source_list:
+            answer = self._resolve_value_source(value_source)
+            if isinstance(answer, list):
+                answers.extend(answer)
+            else:
+                answers.append(answer)
+        return answers
```

Comment:
- This is building up a list of `values` rather than `answers`, as the value source could be metadata.
  ```suggestion
    def _resolve_value_source_list(self, value_source_list: list[Dict]) -> list[str]:
        values: list[str] = []
        for value_source in value_source_list:
            value = self._resolve_value_source(value_source)
            if isinstance(value, list):
                values.extend(value)
            else:
                values.append(value)
        return values
  ```

- The types need to be looked at as the values could be numbers (they could also be dicts e.g. for duration answers, but that wouldn't work here anyway).
- I've had a look but can't see it, is there a test that demonstrates using an answer value source that resolves to a list?

Reply:
Test no, schema yes, https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/test_schemas/en/test_question_title_in_error.json#L105, so perhaps a specific test is appropriate. However I do wonder if we should have a separate transformation for checkboxes concatenating lists. In this instance the transform does not need to be a list, which the story dictates. I can't see why we would combine checkboxes or checkboxes and strings

Reply:
I think the checkbox unit test that you've added is enough. I think we should either leave it as it is, or change the argument to allow an object (for a value source that resolved to a list i.e. a checkbox answer) or an array (for a list of value sources).